src/template_utils.py

```diff
@@ -0,0 +1,56 @@
+import os
+import re
+from typing import Any, Dict
+
+import yaml
+
+
+def load_yaml_template(template_path: str) -> str:
+    """
+    Load a YAML template from a file.
+
+    Args:
+        template_path: Path to the YAML template file
+
+    Returns:
+        The template as a string
+    """
+    with open(template_path, "r") as f:
+        return f.read()
+
+
+def render_template(template: str, values: Dict[str, Any]) -> str:
+    """
+    Render a template by replacing placeholders with values.
+
+    Args:
+        template: Template string with ${PLACEHOLDER} variables
+        values: Dictionary of values to replace placeholders
+
+    Returns:
+        Rendered template as a string
+    """
+    rendered = template
+
+    # Replace all ${PLACEHOLDER} variables
+    for key, value in values.items():
+        placeholder = f"${{{key}}}"
+        rendered = rendered.replace(placeholder, str(value))
+
+    return rendered
```

Comment:
I would suggest using mustache or jinja or another templating library here. This works for now but it's fragile